components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/impersonation/models/ImpersonationRequestDTO.java

```diff
@@ -20,26 +20,27 @@
 package org.wso2.carbon.identity.oauth2.impersonation.models;
 
 import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.model.ImpersonatedUser;
 import org.wso2.carbon.identity.oauth2.authz.OAuthAuthzReqMessageContext;
 
 /**
  * Request DTO class for Impersonation validation.
  */
 public class ImpersonationRequestDTO {
 
-    private String subject;
+    private ImpersonatedUser subject;
     private AuthenticatedUser impersonator;
     private String clientId;
     private String[] scopes;
     private String tenantDomain;
     private OAuthAuthzReqMessageContext oAuthAuthzReqMessageContext;
 
-    public String getSubject() {
+    public ImpersonatedUser getSubject() {
 
         return subject;
     }
 
-    public void setSubject(String subject) {
+    public void setSubject(ImpersonatedUser subject) {
```

Comment:
changing a public method signature, hope you checked other implementations

Reply:
Yes.